test/unit/test_blas_tile/test_hemm.h

```diff
@@ -0,0 +1,150 @@
+  T beta = TypeUtilities<T>::element(1.1, .4);
+  BaseType<T> gamma = 1.3f;
+
+  memory::MemoryView<T, Device::CPU> mem_a(mul(lda, size_a.cols()));
+  memory::MemoryView<T, Device::CPU> mem_b(mul(ldb, size_b.cols()));
+  memory::MemoryView<T, Device::CPU> mem_c(mul(ldc, size_c.cols()));
+
+  // Create tiles.
+  Tile<T, Device::CPU> a0(size_a, std::move(mem_a), lda);
+  Tile<T, Device::CPU> b0(size_b, std::move(mem_b), ldb);
+  Tile<T, Device::CPU> c(size_c, std::move(mem_c), ldc);
+
+  // Note: The tile elements are chosen such that:
+  // Cij = 1.2 * i / (j+1) * exp(I * (j-i))
+  // where I is the imaginary number
+  //
+  // if side == Left
+  // Aik = 0.9 * (i+1) * (k+1) * exp(gamma * I * (i-k))
+  // Bkj = 0.7 / ((j+1) * (k+1)) * exp(gamma * I * (k+j))
+  //
+  // if side == Right
+  // Bik = 0.7 / ((i+1) * (k+1)) * exp(gamma * I * (i+k))
+  // Akj = 0.9 * (j+1) * (k+1) * exp(gamma * I * (j-k))
+  //
+  // Hence the solution (S) will be
+  // if side == Left
+  // Sij = beta * Cij + 0.63 * k * gamma * (i+1)/(j+1) exp(I * alpha * (i+j))
+  // if side == Right
+  // Sij = beta * Cij + 0.63 * k * gamma * (j+1)/(i+1) exp(I * alpha * (i+j))
+  auto el_a = [side, gamma](const TileElementIndex& index) {
+    if (side == blas::Side::Left) {
+      double i = index.row();
+      double k = index.col();
+      return TypeUtilities<T>::polar(.9 * (i + 1) * (k + 1), gamma * (i - k));
+    }
+    else {
+      double k = index.row();
+      double j = index.col();
+      return TypeUtilities<T>::polar(.9 * (j + 1) * (k + 1), gamma * (j - k));
+    }
+  };
+
+  auto el_b = [side, gamma](const TileElementIndex& index) {
+    if (side == blas::Side::Left) {
+      double k = index.row();
+      double j = index.col();
+      return TypeUtilities<T>::polar(.7 / ((j + 1) * (k + 1)), gamma * (k + j));
+    }
+    else {
+      double i = index.row();
+      double k = index.col();
+      return TypeUtilities<T>::polar(.7 / ((i + 1) * (k + 1)), gamma * (i + k));
+    }
+  };
+
+  auto el_c = [](const TileElementIndex& index) {
+    double i = index.row();
+    double j = index.col();
+    return TypeUtilities<T>::polar(1.2 * i / (j + 1), -i + j);
+  };
+
+  auto res_c = [side, k, alpha, beta, gamma, el_c](const TileElementIndex& index) {
+    double i = index.row();
+    double j = index.col();
+
+    if (side == blas::Side::Left) {
+      return beta * el_c(index) + TypeUtilities<T>::element(0.63 * k, 0) * alpha *
+                                      TypeUtilities<T>::polar((i + 1) / (j + 1), gamma * (i + j));
+    }
+    else {
+      return beta * el_c(index) + TypeUtilities<T>::element(0.63 * k, 0) * alpha *
+                                      TypeUtilities<T>::polar((j + 1) / (i + 1), gamma * (i + j));
+    }
+  };
+
+  // Set tile elements.
+  set(a0, el_a);
+  set(b0, el_b);
+  set(c, el_c);
+
+  // Read-only tiles become constant if CT is const T.
+  const Tile<CT, Device::CPU> a(std::move(a0));
+  const Tile<CT, Device::CPU> b(std::move(b0));
```

Comment:
FYI
From what I see, you'd like to initialize readonly `Tile<const T>`, but you have to use a "support" `Tile<T>` (e.g. `a0`), so you can initialize it with the functions above and then you move the support tile to the final read-only one.

An alternative can be doing something like this:

```cpp
auto setup_readonly_tile = [](auto&& values, const auto size, const SizeType ld){
  memory::MemoryView<T, Device::CPU> support_mem(mul(ld, size.cols()));
  Tile<T, Device::CPU> support(size, std::move(support_mem), ld);
  set(support, values);
  return Tile<CT, Device::CPU>(std::move(support));
};

const Tile<CT, Device::CPU> a = setup_readonly_tile(el_a);
const Tile<CT, Device::CPU> b = setup_readonly_tile(el_b);
const Tile<CT, Device::CPU> c = setup_readonly_tile(el_c);
```

*DISCLAIMER* I didn't test it, so I'm not sure it works. But I hope you get the idea.

Reply:
This is used everywhere in the tile blas and lapack tests. Should be addressed in a new PR...